emcc.py

```diff
@@ -876,12 +880,21 @@ def need_llvm_debug_info():
 
     options, settings_changes, user_js_defines, newargs = parse_args(newargs)
 
+    has_dash_c = '-c' in newargs
+    has_dash_S = '-S' in newargs
+    has_dash_E = '-E' in newargs
+
+    compile_only = check_compile_only(newargs)
+
     if options.post_link or options.oformat == OFormat.BARE:
       diagnostics.warning('experimental', '--oformat=base/--post-link are experimental and subject to change.')
 
     if '-print-search-dirs' in newargs:
       return run_process([clang, '-print-search-dirs'], check=False).returncode
 
+    if compile_only and len([x for x in newargs if x.startswith('-sINLINING_LIMIT') or x == 'INLINING_LIMIT' or x.startswith('INLINING_LIMIT=')]):
+      exit_with_error('-s INLINING_LIMIT is a link time setting! (Pass -fno-inline-functions at compile time to disable inlining)')
```

Comment:
I think this might be better of using the existing diagnostic that we have:

See 'unused-command-line-argument' else where in this file:

```
diagnostics.warning('unused-command-line-argument', "...."
```

That why its controllable with the normal `-Werror` system.

Reply:
If I demoted this from an error to a warning, I would then lose the benefit of having the guarantee with ccache that this flag will not be present at compile time.

Reply:
What about adding `-Werrror=unused-command-line-argument` to your builds to turn this into an error?

Alternatively what about putting it behind `STRICT=1`?   It seems like that would be better since it would result in less breakage in the wild.

I still like the idea of having different internal settings for linking vs compiling because we have to handle that case where the we are compiling and linking in the same command.  Would having this fags only effect the linking phase be enough to fix your issue?

Reply:
> What about adding `-Werrror=unused-command-line-argument` to your builds to turn this into an error?

I am implementing general support into ccache project (https://github.com/ccache/ccache) to support Emscripten, so it is not a question of what arguments I am personally passing in my builds.

That being said, I went ahead and implemented these flags to ccache instead, so we don't need this change.